### PR TITLE
Updating build script for libpng 1.6.37

### PR DESCRIPTION
**Pull request by HCLJason**
This is the result of emconfigure, except with PNG warnings disabled to remove junk warnings.

This needs to get edited and merged after the matching emscripten-ports/libpng PR gets merged.

### CODE REVIEW

File: tools/ports/libpng.py [hunk 1]
```diff
@@ -7,8 +7,9 @@
 import shutil
 import logging
 
+# TODO - update these once upstream is current
 TAG = 'version_1'
```
Comment (sbc100) [inline tools/ports/libpng.py:11]: This seems odd that the tag stayed the same but the HASH of the content didn't change?   I would expect this to change to `version_2`, or something other than `version_1`

**Review by sbc100 — APPROVED** [review]
Looks good to me.

Same comment as with zlib about possibly moving the config header out of this file and storing it upstream.

**Comment by HCLJason** [comment]
As discussed, we're switching to use upstream projects instead of emscripten-ports for projects with no Emscripten-specific changes.

The build errors might be CircleCI-related.

system_libs:INFO: retrieving port: libpng from https://download.sourceforge.net/libpng/libpng-1.6.37.tar.gz
system_libs:INFO: unpacking port: libpng
cache:INFO: generating port: sysroot/lib/wasm32-emscripten/libpng.a... (this will be cached in "/local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libpng.a" for subsequent builds)
root:INFO: building port: libpng
cache:INFO:  - ok

File: tools/ports/libpng.py [hunk 2]
```diff
@@ -19,7 +18,7 @@ def needed(settings):
 
 
 def get(ports, settings, shared):
-  ports.fetch_project('libpng', 'https://github.com/emscripten-ports/libpng/archive/' + TAG + '.zip', 'libpng-' + TAG, sha512hash=HASH)
+  ports.fetch_project('libpng', 'https://download.sourceforge.net/libpng/libpng-' + TAG + '.tar.gz', 'libpng-' + TAG, sha512hash=HASH)
```
Comment (sbc100) [inline tools/ports/libpng.py:21]: We have had problems with sourceforge downloads in the past.   I'd rather not rely on their uptime.

The solution we normally have is to mirror the archive on our own google storage bucket.

See #13935 and #13878.

If you would like me upload a certain version of libpng to this bucket just let me know and I can do that.

Comment (HCLJason) [reply tools/ports/libpng.py:21]: Would we be better off updating the emscripten-port's copy, then? It's 6 vs half-dozen.

If you want to store it in a Google bucket, this is the most recent file from the libpng.org site:
https://download.sourceforge.net/libpng/libpng-1.6.37.tar.gz

Comment (sbc100) [reply tools/ports/libpng.py:21]: I uploaded to https://storage.googleapis.com/webassembly/emscripten-ports/libpng-1.6.37.tar.gz.

The SHASUM should be identical.  

I think its nicer to use a mirror of the pristine sources when we have no patches to maintain.    Have a separate github repo makes it look like we have/need patches.

File: tools/ports/libpng.py (hunk 1 above)
Comment (HCLJason) [reply tools/ports/libpng.py]: Fixed indirectly with the Google storage fix.

File: tools/ports/libpng.py (hunk 2 above)
Comment (HCLJason) [reply tools/ports/libpng.py]: I updated the script to use that, but it's still failing in CircleCI.

**Comment by sbc100** [comment]
The errors you CI are not related to your change, they should be fixed by rebasing/merging into/from the `main` branch.  Can you give that a try?

**Pull request closed without merging**